src/common/components/name.cpp

```diff
@@ -0,0 +1,6 @@
+#include "name.h"
+
+cqsp::common::components::Name::Name() {}
+
+cqsp::common::components::Name::Name(const std::string &p_name)
+    : name(p_name) {}
```

Comment:
Put all of this into the header if you *have* to add this, it's not much code, and doesn't justify adding a new file

Also I don't think we need a constructor because entt already does this for us.